# Copyright (c) <2016>, <Matthew Horn>
# All rights reserved.
# Redistribution and use in source and binary forms, with or without modification, 
# are permitted provided that the following conditions are met:
# 1. Redistributions of source code must retain the above copyright notice, this 
# list of conditions and the following disclaimer.
# 2. Redistributions in binary form must reproduce the above copyright notice, 
# this list of conditions and the following disclaimer in the documentation and/or
#  other materials provided with the distribution.
# 3. Neither the name of the copyright holder nor the names of its contributors 
# may be used to endorse or promote products derived from this software without
#  specific prior written permission.

# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS 
# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
# THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

<launch>
  <arg name="sim" default="false" />
  <arg name="node" default="false" />
  <arg name="ip" default="192.168.1.11" />
  <arg name="robotiq_prefix" default="" />

  <node 
    unless="$(arg sim)"
    name="SModelTcpNode"
    pkg="robotiq_s_model_control"
    type="SModelTcpNode.py" 
    args="$(arg ip)"
    output="screen"
  />
  <node 
    if="$(arg sim)"
    name="SModelTcpNode"
    pkg="robotiq_driver"
    type="fake_robotiq.py"
    args=""
    output="screen"
  />
  
  <node
    name="robotiq_driver_node"
    pkg="robotiq_driver"
    type="robotiq_driver_node" 
    args="$(arg robotiq_prefix)"
    output="screen"
  />
  
  <group if="$(arg node)">
    <node
      if="$(arg sim)"
      name="rs_gripper_interface_node"
      pkg="grasp_interface"
      type="rs_gripper_interface_node" 
      args="sim"
      output="screen"
    />
    
    <node
      unless="$(arg sim)"
      name="rs_gripper_interface_node"
      pkg="grasp_interface"
      type="rs_gripper_interface_node" 
      output="screen"
    />
  </group>
</launch>
